src/Microsoft.HttpRepl.Fakes/MockHttpMessageHandler.cs

```diff
@@ -19,8 +20,10 @@ public class MockHttpMessageHandler : HttpMessageHandler
         public MockHttpMessageHandler(IDictionary<string, string> urlsWithResponse,
```

Comment:
This feels weird to me to have a constructor that takes three distinct possibilities of use cases - only one of `urlsWithResponse` OR `header` OR `readFromFile/fileContents` will be used for any given MockHttpMessageHandler instance. I'm guessing this was just written to follow along with the code that calls it, which has all of those options as default parameters. But this class itself doesn't have default parameters, and that three option construct just feels weird. Maybe it doesn't matter right now, since its "just" a mock and we may need to refactor later anyway (for reasons @jimmylewis mentioned in the comment a few minutes ago)? Just wanted to call it out to see what people thought.

Reply:
I agree with this, just didn't have a good way of phrasing it.  Seems like this fake component should really be a `Dictionary<string, HttpResponseMessage>`.  We've moved some of the response message details to the overall MessageHandler type to apply to all responses (but in a unit test, how many responses are you actually issuing anyway?  Shouldn't be too many); and I don't understand the value of the readFromFile/fileContents pairing (why not just provide the response content?).

I think it's worth re-evaluating this fake in a separate PR.